File: FrontEnd-Angular/src/app/services/renda/renda.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';

const BASIC_URL = "http://localhost:8080";

@Injectable({
  providedIn: 'root'
})
export class RendaService {
  constructor(private http: HttpClient) { }

  postRenda(rendaDTO: any): Observable<any> {
    return this.http.post(`${BASIC_URL}/api/renda`, rendaDTO);
  }

  getAllrendas(): Observable<any> {
    return this.http.get(`${BASIC_URL}/api/renda/all`);
  }

  getRendaById(id:number): Observable<any> {
    return this.http.get(`${BASIC_URL}/api/renda/${id}`);
  }

  atualizarRenda(id:number, rendaDTO:any): Observable<any> {
    return this.http.put(`${BASIC_URL}/api/renda/${id}` , rendaDTO);
  }

  deleteRenda(id:number): Observable<any> {
    return this.http.delete(`${BASIC_URL}/api/renda/${id}`);
  }
}

File: Backend/ProjetoFinal/src/main/java/com/Financas/ProjetoFinal/services/Renda/RendaService.java
package com.Financas.ProjetoFinal.services.Renda;


import com.Financas.ProjetoFinal.dto.RendaDTO;
import com.Financas.ProjetoFinal.entity.Renda;

import java.util.List;

public interface RendaService {

    Renda postRenda(RendaDTO rendaDTO);

    List<RendaDTO> getAllRendas();

    Renda updateRenda(Long id, RendaDTO rendaDTO);

    RendaDTO getRendaById(Long id);

    void deleteRenda(Long id);



}
